cli/src/commands/extensions/permissions.rs

```diff
@@ -61,6 +62,65 @@ impl Permission {
             Err(anyhow!("Requires {resource_type} access to {resource:?}"))
         }
     }
+
+    pub fn subset_of(&self, parent: &Permission) -> Result<Permission> {
+        match (parent, self) {
+            // Child deny-all always succeeds, returning deny-all.
+            (_, &Permission::Boolean(false)) => Ok(Permission::Boolean(false)),
+            // Parent deny-all fails with all child permissions but deny-all.
+            (&Permission::Boolean(false), _) => {
+                Err(anyhow!("All permissions are denied by the manifest"))
+            },
+            // Parent allow-all always succeeds, returning the child's permissions.
+            (&Permission::Boolean(true), child) => Ok(child.clone()),
+            // Child allow-all fails with more restrictive parent permissions.
+            (_, &Permission::Boolean(true)) => {
+                Err(anyhow!("The requested permissions are denied by the manifest"))
+            },
+            // Parent set vs child set have to be validated.
+            // This will error if child is not subset of parent, and return the child set otherwise.
+            (&Permission::List(ref parent), &Permission::List(ref child)) => {
+                Permission::check_paths_include_children(parent, child)
+                    .map(|_| Permission::List(child.clone()))
+                    .map_err(|mismatches| {
+                        anyhow!(
+                            "The following paths are denied by the manifest: {}",
+                            mismatches.join(", ")
+                        )
+                    })
+            },
+        }
+    }
+
+    fn check_paths_include_children(
+        parent: &[String],
+        child: &[String],
+    ) -> StdResult<(), Vec<String>> {
+        let parent_paths = parent.iter().map(Path::new).collect::<Vec<_>>();
+        let child_paths = child.iter().map(Path::new).collect::<Vec<_>>();
+
+        // Find all paths in `child` that don't have a prefix in `parent`.
+        let without_parent: Vec<_> = child_paths
+            .iter()
+            .filter_map(|child| {
+                // Using Path::starts_with rather than String::starts_with in order to get
+                // the correct semantics.
+                if parent_paths.iter().any(|p| child.starts_with(p)) {
+                    None
+                } else {
+                    Some(child.to_string_lossy().to_string())
+                }
+            })
+            .collect::<Vec<_>>();
```

Comment:
```suggestion
        // Find all paths in `child` that don't have a prefix in `parent`.
        let without_parent: Vec<_> = child
            .iter()
            .filter_map(|child| {
                // Using Path::starts_with rather than String::starts_with in order to get
                // the correct semantics.
                if parent.iter().any(|p| Path::new(child).starts_with(p)) {
                    None
                } else {
                    Some(child.clone())
                }
            })
            .collect::<Vec<_>>();
```

Reply:
Oh, yeah, this is a great point indeed. I like it.